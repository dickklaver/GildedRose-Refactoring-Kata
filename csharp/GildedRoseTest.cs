using ApprovalTests;
using ApprovalTests.Combinations;
using ApprovalTests.Reporters;

//using ApprovalUtilities;

using NUnit.Framework;

using System.Collections.Generic;
using System.Linq;

namespace csharp
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class GildedRoseTest
    {
        private IEnumerable<string> names;
        private IEnumerable<int> sellIns;
        private IEnumerable<int> qualities;

        public GildedRoseTest()
        {
            names = new[] { "foo", "bar", "baz" };
            sellIns = Enumerable.Range(-2, 15);
            qualities = Enumerable.Range(-2, 15);
        }
        [Test]
        public void foo()
        {
            CombinationApprovals.VerifyAllCombinations(TestFoo, names, sellIns, qualities);
        }

        [Test]
        public void brie()
        {
            CombinationApprovals.VerifyAllCombinations(TestBrie, sellIns, qualities);
        }

        [Test]
        public void backstage()
        {
            CombinationApprovals.VerifyAllCombinations(TestBackstage, sellIns, qualities);
        }

        [Test]
        public void conjured()
        {
            CombinationApprovals.VerifyAllCombinations(TestConjured, names, sellIns, qualities);
        }

        [Test]
        public void sulfuras()
        {
            CombinationApprovals.VerifyAllCombinations(TestSulfuras, sellIns);
        }

        private string TestFoo(string name, int sellIn, int quality)
        {
            IList<Item> Items = new List<Item> { new GeneralItem(name, sellIn, quality) };
            return TestIt(Items);
        }

        private string TestBrie(int sellIn, int quality)
        {
            IList<Item> Items = new List<Item> { new AgedBrie(sellIn, quality) };
            return TestIt(Items);
        }

        private string TestBackstage(int sellIn, int quality)
        {
            IList<Item> Items = new List<Item> { new BackstagePass(sellIn, quality) };
            return TestIt(Items);
        }

        private string TestConjured(string name, int sellIn, int quality)
        {
            IList<Item> Items = new List<Item> { new Conjured(name, sellIn, quality) };
            return TestIt(Items);
        }

        private string TestSulfuras(int sellIn)
        {
            IList<Item> Items = new List<Item> { new Sulfuras(sellIn) };
            return TestIt(Items);
        }

        private static string TestIt(IList<Item> Items)
        {
            GildedRose app = new GildedRose(Items);
            app.UpdateQuality();
            return Items[0].ToString();
        }
    }
}
