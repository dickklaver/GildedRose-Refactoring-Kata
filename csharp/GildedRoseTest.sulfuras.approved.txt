[-2] => Sulfuras, Hand of Ragnaros, -2, 80
[-1] => Sulfuras, Hand of Ragnaros, -1, 80
[0] => Sulfuras, Hand of Ragnaros, 0, 80
[1] => Sulfuras, Hand of Ragnaros, 1, 80
[2] => Sulfuras, Hand of Ragnaros, 2, 80
[3] => Sulfuras, Hand of Ragnaros, 3, 80
[4] => Sulfuras, Hand of Ragnaros, 4, 80
[5] => Sulfuras, Hand of Ragnaros, 5, 80
[6] => Sulfuras, Hand of Ragnaros, 6, 80
[7] => Sulfuras, Hand of Ragnaros, 7, 80
[8] => Sulfuras, Hand of Ragnaros, 8, 80
[9] => Sulfuras, Hand of Ragnaros, 9, 80
[10] => Sulfuras, Hand of Ragnaros, 10, 80
[11] => Sulfuras, Hand of Ragnaros, 11, 80
[12] => Sulfuras, Hand of Ragnaros, 12, 80
